<?xml version="1.0" ?>
<launch>
    <arg name="global_frame" default="/world" />
    <arg name="tool_points" default="3" />
    <arg name="load_reference" default="false" />

    <!-- Lenght of the adapter attached on the tip of EGS -->
    <arg name="adapter_len" default="0.15" />

    <node pkg="global_frame_registration" type="adapter_pose_publisher" name="adapter_pose_publisher" output="screen" unless="$(arg load_reference)">
        <param name="adapter_len" value="$(arg adapter_len)"/>
    </node>

    <include file="$(find global_frame_registration)/launch/board_recognition.launch" />

    <node pkg="global_frame_registration" name="global_frame_registration" type="global_frame_registration" required="true" output="screen">
        <param name="global_frame_name" type="string" value="$(arg global_frame)" />
        <param name="num_points" type="int" value="$(arg tool_points)" />
        <param name="use_existing" type="bool" value="$(arg load_reference)" />
        <param name="eye_in_hand" type="bool" value="true"/>
        <param name="camera_color_frame" value="camera_color_optical_frame"/>
        <param name="camera_ref_frame" value="ECM"/>
        <param name="camera_ref_base_frame" value="ECM_base"/>

        <rosparam>
        rgbd_subscribers: [/realsense_pose_array]
        tool_subscribers: [/dvrk/ECM/adapter_position] 
        camera_subscribers: []
        registrations:
        - frame_id: camera_color_optical_frame
          orientation: [-0.0054425456418275555, -0.005027710385426786, 0.030487809301417177, 0.99950767596313]
          parent_frame: ECM
          position: [-0.03500508786250414, 0.01967764025743594, -0.03261519141243981]
        - frame_id: ECM_base
          orientation: [0.0218182868388352, 0.03965589296110484, 0.6883086762891213, 0.7240045156337033]
          parent_frame: /world
          position: [0.11254918368379721, 0.011677777824120213, 0.3221577593982077]
        </rosparam>
    </node>


</launch>